src/main/java/com/powsybl/openloadflow/sa/AbstractSecurityAnalysis.java

```diff
@@ -241,4 +243,57 @@ protected static void findAllSwitchesToOperate(Network network, List<Action> act
                     }
                 });
     }
+
+    protected Optional<OperatorStrategyResult> runActionSimulation(LfNetwork network, LoadFlowContext context, OperatorStrategy operatorStrategy,
+                                                                 LimitViolationManager preContingencyLimitViolationManager,
+                                                                 SecurityAnalysisParameters.IncreasedViolationsParameters violationsParameters,
+                                                                 LimitViolationsResult postContingencyLimitViolations, Map<String, LfAction> lfActionById,
+                                                                 boolean createResultExtension, LfContingency contingency, boolean dc) {
+        OperatorStrategyResult operatorStrategyResult = null;
+
+        if (checkCondition(operatorStrategy, postContingencyLimitViolations)) {
```

Comment:
The condition is checked twice for DC security analysis, no?

Reply:
For sure you cannot apply the contingency if not relevant and restore the network. It is a lost of performance and it increases the probability of errors. The mutualisation cannot be done despide that points.

Reply:
I refactor the code to only check violation one time in DC SA and keep as much code as possible in common. It should be ok now.